component/prometheus/scrape/scrape.go

```diff
@@ -91,27 +91,26 @@ type Clustering struct {
 	Enabled bool `river:"enabled,attr"`
 }
 
-// DefaultArguments defines the default settings for a scrape job.
-var DefaultArguments = Arguments{
-	MetricsPath:      "/metrics",
-	Scheme:           "http",
-	HonorLabels:      false,
-	HonorTimestamps:  true,
-	HTTPClientConfig: component_config.DefaultHTTPClientConfig,
-	ScrapeInterval:   1 * time.Minute,  // From config.DefaultGlobalConfig
-	ScrapeTimeout:    10 * time.Second, // From config.DefaultGlobalConfig
+// SetToDefault implements river.Defaulter.
+func (arg *Arguments) SetToDefault() {
+	*arg = Arguments{
+		MetricsPath:      "/metrics",
+		Scheme:           "http",
+		HonorLabels:      false,
+		HonorTimestamps:  true,
+		HTTPClientConfig: component_config.DefaultHTTPClientConfig,
+		ScrapeInterval:   1 * time.Minute,  // From config.DefaultGlobalConfig
+		ScrapeTimeout:    10 * time.Second, // From config.DefaultGlobalConfig
+	}
 }
 
 // UnmarshalRiver implements river.Unmarshaler.
 func (arg *Arguments) UnmarshalRiver(f func(interface{}) error) error {
-	*arg = DefaultArguments
-
-	type args Arguments
-	err := f((*args)(arg))
-	if err != nil {
-		return err
-	}
+	return f(arg)
+}
```

Comment:
I realise it's a draft - just a reminder to remove the entire UnmarshalRiver here - AFAICT we won't need it.

Reply:
removed